plugin/pkg/auth/authorizer/node/metrics.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const nodeAuthorizerSubsystem = "node_authorizer"
+
+var (
+	graphActionsDuration = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Subsystem:      nodeAuthorizerSubsystem,
+			Name:           "graph_actions_duration_seconds",
+			Help:           "Histogram of duration of graph actions in node authorizer.",
+			StabilityLevel: metrics.ALPHA,
+			// Start with 0.1ms with the last bucket being [~200ms, Inf)
+			Buckets: metrics.ExponentialBuckets(0.0001, 2, 12),
+		},
+		[]string{"operation"},
+	)
+)
+
+func init() {
```

Comment:
Would you mind modifying this to the following signature?

```golang
func RegisterMetrics(r Registry) {
    r.MustRegister(graphActionsDuration)
}
```

And then instead of importing this package to instantiate the metric as a side-effect, we can explicitly invoke the MustRegister function with the metric registry in some boot sequence code-path. This is the pattern we are trying to move towards. We do this currently in the kubelet (https://github.com/kubernetes/kubernetes/blob/v1.19.0-alpha.0/pkg/kubelet/server/server.go#L368-L373). 

You can also just pass in the legacy registry instead of a custom one, if your metrics endpoint is already just using the legacyregistry. This will make it easier for us to get rid of the legacyregistry later on (which we intend to do, since it is deprecated).

Reply:
I changed to this pattern with one diff: legancyregistry is a package and not an object, so I cannot call RegisterMetrics(legacyregistry). I changed the signature to accept function to call, like: RegisterMetrics(legacyregistry.MustRegister). Please let me know if this works.

Reply:
It's not possible to do that this way with legacyregistry:
* I added a node.RegisterMetrics(legacyregistry.MustRegistry) call to node/config (where we decide if we are going to use node authorizer)
* This node/config can be called multiple times in a single binary (e.g. in our tests)
* This triggers panic due to multiple metric registration
* I have to add sync.Once to dedupe metrics registration, but this way, if RegisterMetric will be called twice for multiple metric registries, the second call will be completely ignored leading to not registering this
* I changed this to the pattern that is used everywhere in kube-apiserver for legacyregistry: hardcoding legacyregistry in MustRegister and calling MustRegister on the node/config which seems to be the best option we have currently in kube-apiserver.

Reply:
Yeah, it's a bit entangled in the apiserver currently.